src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package org.example.bookshop.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.example.bookshop.dto.CartItemDto;
+import org.example.bookshop.dto.CartItemRequestDto;
+import org.example.bookshop.dto.CartUpdateRequestDto;
+import org.example.bookshop.dto.ShoppingCartDto;
+import org.example.bookshop.exception.EntityNotFoundException;
+import org.example.bookshop.mapper.CartItemMapper;
+import org.example.bookshop.mapper.ShoppingCartMapper;
+import org.example.bookshop.model.Book;
+import org.example.bookshop.model.CartItem;
+import org.example.bookshop.model.ShoppingCart;
+import org.example.bookshop.model.User;
+import org.example.bookshop.repository.BookRepository;
+import org.example.bookshop.repository.CartItemRepository;
+import org.example.bookshop.repository.CartRepository;
+import org.example.bookshop.service.CartService;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CartServiceImpl implements CartService {
+    private final BookRepository bookRepository;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final CartRepository cartRepository;
+    private final ShoppingCartMapper cartMapper;
+
+    @Override
+    public ShoppingCartDto get(User user) {
+        ShoppingCart shoppingCart = findShoppingCart(user);
+        return cartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public CartItemDto add(CartItemRequestDto requestDto, User user) {
+        CartItem cartItem = cartItemMapper.toCartItem(requestDto);
```

Comment:
it's better to find book, find shop cart, check if cart already include the same book, if yes - increment quantity, if no - add book to cart

It's better to save shop cart at the end of method